packages/webdriverio/src/commands/browser/waitUntil.js

```diff
@@ -0,0 +1,79 @@
+/**
+ *
+ * This wait command is your universal weapon if you want to wait on something. It expects a condition
+ * and waits until that condition is fulfilled with a truthy value. If you use the WDIO testrunner the
+ * commands within the condition are getting executed synchronously like in your test.
+ *
+ * A common example is to wait until a certain element contains a certain text (see example).
+ *
+ * <example>
+    :example.html
+    <div id="someText">I am some text</div>
+    <script>
+      setTimeout(function() {
+        $('#someText').html('I am now different');
+      }, 1000);
+    </script>
+
+    :waitUntil.js
+    it('should wait until text has changed', function () {
+        browser.waitUntil(function () {
+          return browser.getText('#someText') === 'I am now different'
+        }, 5000, 'expected text to be different after 5s');
+    });
+ * </example>
+ *
+ *
+ * @alias browser.waitUntil
+ * @param {Function} condition  condition to wait on
+ * @param {Number=}  timeout    timeout in ms (default: 500)
+ * @param {String=}  timeoutMsg error message to throw when waitUntil times out
+ * @param {Number=}  interval   interval between condition checks (default: 500)
+ * @uses utility/pause
+ * @type utility
+ *
+ */
+
+// TODO - Reimplement these when ErrorHandler has been ported
```

Comment:
Let's remove these remarks, I am not sure if it makes sense to port these error types to v5

Reply:
Are you suggesting we would simply throw generic errors?  If so I'm not sure I agree with this approach.  Specific error types are useful when trying to analyze / categorize test failures

Reply:
true .. let's remove these todos for now tho

Reply:
Done.  Logged #9 so we don't forget